pkg/helm/helm.go

```diff
@@ -93,6 +93,34 @@ func InstallOrUpgradeHelmChartFromPlugin(ctx context.Context, local client.Clien
 	return upgradeRelease(ctx, local, restClientGetter, pluginDefinition, plugin)
 }
 
+// HelmChartTest to do helm test on the plugin
+func HelmChartTest(ctx context.Context, restClientGetter genericclioptions.RESTClientGetter, plugin *greenhousev1alpha1.Plugin) (bool, error) {
+	var hasTestHook bool
+	cfg, err := newHelmAction(restClientGetter, plugin.GetReleaseNamespace())
+	if err != nil {
+		return hasTestHook, err
+	}
+
+	results, err := action.NewReleaseTesting(cfg).Run(plugin.Name)
+	if err != nil {
+		return hasTestHook, err
+	}
+
+	if results.Hooks != nil {
+	outer:
+		for _, hook := range results.Hooks {
+			for _, event := range hook.Events {
+				if event == release.HookTest {
+					hasTestHook = true
+					break outer // Break out of both loops when a test hook is found
+				}
+			}
+		}
```

Comment:
NIT: What do you think about using the _slices_ package instead?

The `slices.ContainsFunc(..)` returns on the first hit, also the `slices.Contains(..)` returns on the first hit.

```suggestion
		hasTestHook = slices.ContainsFunc(results.Hooks, func(h *release.Hook) bool {
			return slices.Contains(h.Events, release.HookTest)
		})
```

Reply:
thank you for the suggestion. Implemented in this https://github.com/cloudoperators/greenhouse/pull/441 commit.